tests/system/src/test/java/test/javafx/stage/FocusedWindowTest.java

```diff
@@ -0,0 +1,127 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.stage;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.scene.control.TextField;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import junit.framework.Assert;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+public class FocusedWindowTest {
+
+    static CountDownLatch startupLatch;
+    static Stage stage = null;
+
+    static {
+        System.setProperty("glass.platform","Monocle");
+        System.setProperty("monocle.platform","Headless");
+    }
+
+    public static class TestApp extends Application {
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            primaryStage.setTitle("Primary Stage");
+            primaryStage.setScene(new Scene(new TextField()));
+
+            primaryStage.setOnShown(l -> {
+                Platform.runLater(() -> startupLatch.countDown());
+            });
+            primaryStage.show();
+            Platform.setImplicitExit(false);
+            stage = primaryStage;
+        }
+    }
+
+    @BeforeClass
+    public static void initFX() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        new Thread(() -> Application.launch(TestApp.class, (String[]) null)).start();
+        Assert.assertTrue("Timeout waiting for FX runtime to start", startupLatch.await(15, TimeUnit.SECONDS));
+        Platform.runLater(() -> stage.close());
+    }
+
+    static WeakReference<Stage> closedFocusedStageWeak = null;
+    static Stage closedFocusedStage = null;
+
+    @Test
+    public void testLeak() throws Exception {
+        Util.runAndWait(() -> {
+            closedFocusedStage = new Stage();
+            closedFocusedStage.setTitle("Focused Stage");
+            closedFocusedStageWeak = new WeakReference<>(closedFocusedStage);
+            TextField textField = new TextField();
+            closedFocusedStage.setScene(new Scene(textField));
+        });
+        Util.runAndWait(() -> {
+            closedFocusedStage.show();
+        });
+        Util.runAndWait(() -> {
+            closedFocusedStage.close();
+        });
+        Util.runAndWait(() -> {
+            closedFocusedStage.requestFocus();
+        });
+        closedFocusedStage = null;
+        assertCollectable(closedFocusedStageWeak);
+    }
```

Comment:
The current test does not assure that stage is shown when `close()` is called or it is closed when `requestFocus()` is called, I would recommend the test to be as, 
```

    @Test
    public void testClosedFocusedStageLeak() throws Exception {
        CountDownLatch latch = new CountDownLatch(1);
        Util.runAndWait(() -> {
            closedFocusedStage = new Stage();
            closedFocusedStage.setTitle("Focused Stage");
            closedFocusedStageWeak = new WeakReference<>(closedFocusedStage);
            TextField textField = new TextField();
            closedFocusedStage.setScene(new Scene(textField));
            closedFocusedStage.setOnShown(l -> {
                latch.countDown();
            });
            closedFocusedStage.show();
        });
        Assert.assertTrue("Timeout waiting for closedFocusedStage to show`",
                      latch.await(15, TimeUnit.MILLISECONDS));

        CountDownLatch hideLatch = new CountDownLatch(1);
        closedFocusedStage.setOnHidden(a -> {
            hideLatch.countDown();
        });
        Util.runAndWait(() -> closedFocusedStage.close());
        Assert.assertTrue("Timeout waiting for closedFocusedStage to hide`",
                      hideLatch.await(15, TimeUnit.MILLISECONDS));

        closedFocusedStage.requestFocus();
        closedFocusedStage = null;
        assertCollectable(closedFocusedStageWeak);
    }
```

Reply:
I've changed it accordingly. I've also verified that the new version catches the old leak and works with my suggested changes.